presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationCompiler.java

```diff
@@ -336,4 +435,9 @@ public static int findAggregationStateParamId(Method method, int id)
         }
         return methods.build();
     }
+
+    static boolean isAggregationMetaAnnotation(Annotation annotation)
```

Comment:
I think filtering anything but @SqlType may cause either unintentional or unexpected change in the behaviour that will be hard to find in the future.
I would say we may wan't to move `BlockIndex` and `AggregationState` directly into isMetaParameter. I didn't do that in the first place not to lose additional verification that someone would not use Aggregation related annotations in scalars.
Also in the further commit, I renamed `isMetaParameter` to `isImplementationDependencyAnnotation` to make naming more to the point.

To summarise - I would keep it as is if you don't have anything against that.